tests/playgrounds/advanced_playground.html

```diff
@@ -0,0 +1,548 @@
+<script src="../../generators/dart/loops.js"></script>
+<script src="../../generators/dart/math.js"></script>
+<script src="../../generators/dart/text.js"></script>
+<script src="../../generators/dart/lists.js"></script>
+<script src="../../generators/dart/colour.js"></script>
+<script src="../../generators/dart/variables.js"></script>
+<script src="../../generators/dart/variables_dynamic.js"></script>
+<script src="../../generators/dart/procedures.js"></script>
+<script src="../../msg/messages.js"></script>
+<script src="../../blocks/logic.js"></script>
+<script src="../../blocks/loops.js"></script>
+<script src="../../blocks/math.js"></script>
+<script src="../../blocks/text.js"></script>
+<script src="../../blocks/lists.js"></script>
+<script src="../../blocks/colour.js"></script>
+<script src="../../blocks/variables.js"></script>
+<script src="../../blocks/variables_dynamic.js"></script>
+<script src="../../blocks/procedures.js"></script>
+<script src="../blocks/test_blocks.js"></script>
+<script src="../themes/test_themes.js"></script>
+<script src="./screenshot.js"></script>
+<script src="https://unpkg.com/@blockly/dev-tools@0.20200313.7/dist/index.js"></script>
+<script src="https://unpkg.com/@blockly/theme-modern@1.20200417.5/dist/index.js"></script>
+
+<script>
+// // Custom requires for the playground.
+// // Rendering.
+goog.require('Blockly.minimalist.Renderer');
+goog.require('Blockly.Themes.Zelos');
+
+// Other.
+goog.require('Blockly.WorkspaceCommentSvg');
+goog.require('Blockly.WorkspaceCommentSvg.render');
+</script>
+<script>
+'use strict';
+var workspace = null;
+
+function start() {
+  setBackgroundColour();
+
+  // Parse the URL arguments.
+  var toolbox = getToolboxElement();
+  var autoimport = !!location.search.match(/autoimport=([^&]+)/);
+  var match = location.search.match(/renderer=([^&]+)/);
+
+  var options = {
+        comments: true,
+        collapse: true,
+        disable: true,
+        grid:
+          {
+            spacing: 25,
+            length: 3,
+            colour: '#ccc',
+            snap: true
+          },
+        horizontalLayout: false,
+        maxBlocks: Infinity,
+        maxInstances: {'test_basic_limit_instances': 3},
+        maxTrashcanContents: 256,
+        media: '../../media/',
+        oneBasedIndex: true,
+        readOnly: false,
+        rtl: false,
+        move: {
+          scrollbars: true,
+          drag: true,
+          wheel: false,
+        },
+        toolbox: toolbox,
+        toolboxPosition: 'start',
+        renderer: 'geras',
+        zoom:
+          {
+            controls: true,
+            wheel: true,
+            startScale: 1.0,
+            maxScale: 4,
+            minScale: 0.25,
+            scaleSpeed: 1.1
+          }
+  };
+
+  // Create main workspace.
+  addGUIControls((options) => {
+    workspace = Blockly.inject('blocklyDiv', options);
+    workspace.configureContextMenu = configureContextMenu;
```

Comment:
I would be in favour of pulling this out into a "createWorkspace" function along side the rest of the functions in here.

Reply:
Done